### PR TITLE
feat: Drop Xcode9 support

### PR DESCRIPTION
**Pull request by mykola-mokhnach**
BREAKING CHANGE: The minimum supported Xcode SDK is bumped to 10.0

So now we can again enable proper warnings handling and get rid of some ugly workarounds. The next step would be to run tests for minimum and maximum supported platforms on Travis (10.0 - 11.x)

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -26,98 +26,100 @@ branches:
...
+# TODO: Test on the minimum and maximum supported platform versions
+
 jobs:
   include:
     - stage: Node tests
-      osx_image: xcode10.2
+      osx_image: xcode10
       language: node_js
-      node_js: "10"
+      node_js: "12"
```
Comment (imurchie) [inline .travis.yml]: Why bump this? 10 is still supported.

File: .travis.yml [hunk 2]
```diff
@@ -26,98 +26,100 @@ branches:
   only:
     - master
 
+# TODO: Test on the minimum and maximum supported platform versions
+
 jobs:
   include:
     - stage: Node tests
-      osx_image: xcode10.2
+      osx_image: xcode10
```
Comment (imurchie) [inline .travis.yml:34]: This isn't needed if it matches the overall version (from line 4).

File: .travis.yml [hunk 3]
```diff
@@ -1,7 +1,7 @@
 language: objective-c
 sudo: false
 os: osx
-osx_image: xcode9.2
+osx_image: xcode10
```
Comment (imurchie) [inline .travis.yml:4]: Why 10 and not 10.3?

Comment (mykola-mokhnach) [reply .travis.yml:4]: We need to make sure WDA works with any minor version including 10.0

File: .travis.yml (hunk 1 above)
Comment (mykola-mokhnach) [reply .travis.yml]: to change less in the future:)

Comment (mykola-mokhnach) [reply .travis.yml]: v12 is also LTS

File: .travis.yml (hunk 2 above)
Comment (mykola-mokhnach) [reply .travis.yml:34]: OK, I'll remove that for tests, which are supposed to run on the minimum supported version

Comment (imurchie) [reply .travis.yml:34]: Makes it easier for the future.

File: .travis.yml (hunk 1 above)
Comment (imurchie) [reply .travis.yml]: But Appium still has to support 10. Given that it changes once a year, seems like a small change to have to do.

File: WebDriverAgentLib/Categories/XCElementSnapshot+FBHitPoint.m [hunk 4]
```diff
@@ -8,47 +8,21 @@
...
 
 @implementation XCElementSnapshot (FBHitPoint)
 
-static BOOL FBHasHitPointProperty = NO;
-static BOOL FBHasHitPointResult = NO;
-static dispatch_once_t onceHitPoint;
-
 - (NSValue *)fb_hitPoint
 {
-  dispatch_once(&onceHitPoint, ^{
```
Comment (KazuCocoa) [inline WebDriverAgentLib/Categories/XCElementSnapshot+FBHitPoint.m:21]: 👍

File: WebDriverAgent.xcodeproj/project.pbxproj [hunk 5]
```diff
@@ -3415,17 +3473,42 @@
 					"$(PROJECT_DIR)/Carthage/Build/Mac",
 					"$(PROJECT_DIR)/Carthage/Build/tvOS",
 				);
-				GCC_TREAT_WARNINGS_AS_ERRORS = NO;
+				GCC_TREAT_WARNINGS_AS_ERRORS = YES;
```
Comment (KazuCocoa) [inline WebDriverAgent.xcodeproj/project.pbxproj:3470]: 👍

**Review by imurchie — APPROVED** [review]

File: .travis.yml (hunk 1 above)
Comment (mAiNiNfEcTiOn) [reply .travis.yml]: Bear in mind you can build using multiple versions of node ;)
Also, v12 will be LTS, for now it's current

File: .travis.yml [hunk 6]
```diff
@@ -26,98 +26,86 @@ branches:
...
-      name: iPhone SE, Xcode 9.2
-      osx_image: xcode9.2
-      env: IPHONE_MODEL="iPhone SE" IPAD_MODEL="iPad Air 2" IOS_VERSION="11.2" ACTION=build TARGET=runner
-    - name: iPhone X, Xcode 10.2
-      osx_image: xcode10.2
-      env: IPHONE_MODEL="iPhone X" IPAD_MODEL="iPad Air 2" IOS_VERSION="12.2" ACTION=build TARGET=runner
-    - name: apple tv, Xcode 10.2
-      osx_image: xcode10.2
-      env: DEST=tv TV_MODEL="Apple TV" TV_VERSION="12.2" ACTION=build TARGET=tv_runner SDK=tv_sim
+      # - name: iPhone SE, Xcode 9.2
```
Comment (KazuCocoa) [inline .travis.yml:41]: just commenting out?

Comment (mykola-mokhnach) [reply .travis.yml:41]: yep, we'll replace them with Xcode 11/iOS 13 as soon as your PR is done

Comment (KazuCocoa) [reply .travis.yml:41]: 👍

**Review by KazuCocoa — APPROVED** [review]

**Pull request merged**